retrofit/src/main/java/retrofit2/OkHttpCall.java

```diff
@@ -113,6 +118,7 @@ private okhttp3.Call getRawCall() throws IOException {
   }
 
   @Override
+  @SuppressWarnings("required.method.not.called") //Response<T> response does not have "close" called on any of its resources, therefore this is a resource leak.
```

Comment:
The scope of this warning suppression could be reduced to the `onResponse` method inside the inner class.

Reply:
I think the intended contract here is for the callback to take ownership of the response, which doesn't happen on errors (leading to a possible leak). Is that correct?